hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -81,6 +82,19 @@ public SCMContainerPlacementRackScatter(final NodeManager nodeManager,
     this.metrics = metrics;
   }
 
+  /**
+   * Constructor for Pipeline Provider Pipeline Placement with rack awareness.
+   * @param nodeManager Node Manager
+   * @param stateManager State Manager
+   * @param conf Configuration
+   */
+  public SCMContainerPlacementRackScatter(NodeManager nodeManager,
+      PipelineStateManager stateManager, ConfigurationSource conf) {
```

Comment:
`PipelineStateManager` currently is unused in `SCMContainerPlacementRackScatter`.

It is used in `PipelinePlacementPolicy` for counting pipelines in individual datanodes so that it can choose a datanode with minimum existing pipelines.  We should do the same as a further improvement.